x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.test.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { shallow, ShallowWrapper } from 'enzyme';
+import { EuiPanel } from '@elastic/eui';
+
+import { ResultField } from './result_field';
+import { ResultHeader } from './result_header';
+import { Result } from './result';
+
+describe('Result', () => {
+  const props = {
+    result: {
+      id: {
+        raw: '1',
+      },
+      title: {
+        raw: 'A title',
+      },
+      description: {
+        raw: 'A description',
+      },
+      length: {
+        raw: 100,
+      },
+      _meta: {
+        id: '1',
+        scopedId: '1',
+        score: 100,
+        engine: 'my-engine',
+      },
+    },
+  };
+
+  it('renders', () => {
+    const wrapper = shallow(<Result {...props} />);
+    expect(wrapper.find(EuiPanel).exists()).toBe(true);
+  });
+
+  it('should render a ResultField for each field except id and _meta', () => {
+    const wrapper = shallow(<Result {...props} />);
+    expect(wrapper.find(ResultField).map((rf) => rf.prop('field'))).toEqual([
+      'title',
+      'description',
+      'length',
+    ]);
+  });
+
+  it('passes through showScore and resultMeta to ResultHeader', () => {
+    const wrapper = shallow(<Result {...props} showScore={true} />);
+    expect(wrapper.find(ResultHeader).prop('showScore')).toBe(true);
+    expect(wrapper.find(ResultHeader).prop('resultMeta')).toEqual({
+      id: '1',
+      scopedId: '1',
+      score: 100,
+      engine: 'my-engine',
+    });
+  });
+
+  describe('when there are more than 5 fields', () => {
+    const propsWithMoreFields = {
+      result: {
+        id: {
+          raw: '1',
+        },
+        title: {
+          raw: 'A title',
+        },
+        description: {
+          raw: 'A description',
+        },
+        length: {
+          raw: 100,
+        },
+        states: {
+          raw: ['Pennsylvania', 'Ohio'],
+        },
+        visitors: {
+          raw: 1000,
+        },
+        size: {
+          raw: 200,
+        },
+        _meta: {
+          id: '1',
+          scopedId: '1',
+          score: 100,
+          engine: 'my-engine',
+        },
+      },
+    };
+
+    describe('the initial render', () => {
```

Comment:
😍 love this test block! so explicit & clear and also super performant (something on my mind as we port more code and our combined plugin tests start taking several minutes to fully run though)